subsys/net/lib/lwm2m/Kconfig

```diff
@@ -43,6 +43,28 @@ config LWM2M_SHELL
 config LWM2M_DTLS_SUPPORT
 	bool "DTLS support in the LwM2M client"
 
+choice
+	prompt "LwM2M DTLS Security type"
+	depends on LWM2M_DTLS_SUPPORT
+	help
+	  Select which security type should be used.
+
+config LWM2M_DTLS_SECURITY_PSK
+	bool "Use Pre-Shared key"
+
+config LWM2M_DTLS_SECURITY_CERTIFICATE
+	bool "Use certificate"
+
+endchoice
+
+config LWM2M_CHECK_SERVER_CERTIFICATE
+	bool "Check returned CA server certificate"
+	depends on LWM2M_DTLS_SECURITY_CERTIFICATE
+	default y
+	help
+	  Check CA of server's certificate. Useful to test Leshan.
+	  Do not use in production mode.
```

Comment:
I think this is a bit confusing, on one hand we have a config that should be unquestionably enabled in any case other than debugging, on other, the help message says `Do not use in production mode.`.

I think it'd be more intuitive if the logic here was reversed, i. e. provide a config, that if enabled, disabler server certificate verification.